custom_components/tuya_local/devices/dr_heater_dr008_thermostat.yaml

```diff
@@ -0,0 +1,229 @@
+          constraint: mode
+          conditions:
+            - dps_val: "Smart"
+              value: auto
+            - dps_val: "Manual"
+              value: heat
+            - dps_val: "Anti_frozen"
+              value: cool
+    - id: 102
+      type: integer
+      name: temperature
+      range:
+        min: 200
+        max: 500
+      mapping:
+        - scale: 10
+          step: 5
+          constraint: temperature_unit
+          conditions:
+            - dps_val: true
+              step: 10
+              range:
+                min: 680
+                max: 1220
+    - id: 103
+      type: string
+      name: mode
+      hidden: true
+    - id: 105
+      type: integer
+      name: current_temperature
+      mapping:
+        - constraint: sensor
+          conditions:
+            - dps_val: "0"
+              scale: 10
+            - dps_val: "1"
+              value_redirect: floor_temperature
+            - dps_val: "2"
+              scale: 10
+    - id: 106
+      type: integer
+      name: floor_temperature
+      hidden: true
+      mapping:
+        - scale: 10
+    - id: 107
+      name: temperature_unit
+      type: boolean
+      mapping:
+        - dps_val: false
+          value: C
+        - dps_val: true
+          value: F
+    - id: 111
+      type: string
+      name: sensor
+      hidden: true
+    - id: 118
+      type: boolean
+      name: hvac_action
+      mapping:
+        - dps_val: true
+          value: heating
+        - dps_val: false
+          value: idle
+    - id: 112
+      type: integer
+      name: temperature_calibration_offset
+      range:
+        min: -9
+        max: 9
+    - id: 113
+      type: integer
+      name: floor_calibration_offset
+      range:
+        min: -9
+        max: 9
+secondary_entities:
+  - entity: lock
+    name: Screen lock
+    category: config
+    dps:
+      - id: 108
+        type: boolean
+        name: lock
+        mapping:
+          - dps_val: true
+            icon: "mdi:hand-back-right-off"
+          - dps_val: false
+            icon: "mdi:hand-back-right"
+  - entity: sensor
+    name: Room temperature
+    class: temperature
+    icon: "mdi:home-thermometer"
+    dps:
+      - id: 105
+        type: integer
+        name: sensor
+        class: measurement
+        mapping:
+          - scale: 10
+      - id: 107
+        name: unit
+        type: boolean
+        mapping:
+          - dps_val: false
+            value: C
+          - dps_val: true
+            value: F
+  - entity: sensor
+    name: Floor temperature
+    class: temperature
+    icon: "mdi:thermometer-lines"
+    dps:
+      - id: 106
+        type: integer
+        name: sensor
+        class: measurement
+        mapping:
+          - scale: 10
+      - id: 107
+        name: unit
+        type: boolean
+        mapping:
+          - dps_val: false
+            value: C
+          - dps_val: true
+            value: F
+  - entity: sensor
+    name: Energy
```

Comment:
If this reliably reports useful data, you can use `class: energy` with `class: total_increasing` on the sensor dp below, then this will be available for tracking in the Energy dashboard.
If it doesn't report useful data, then probably it should just be an attribute of another entity rather than an entity of its own.

Reply:
Most of this was copied from [Jiahong ET-72W thermostat](https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/jiahong_et72w_thermostat.yaml), which was the closest match I was able to find. I only tweaked the DPS values, so I'm not entirely sure how much of this is accurate.

Reply:
Just check what is actually output on that sensor, and whether it is something useful or not. Jianhong is an older device config, probably pre-dating the Energy dashboard in HA, so I don't know whether that one is not marked as energy class because it was unreliable, or it is just not updated.

Reply:
According to the original author, the Jianhong device actually reports minutes of runtime on that dp, and it is multiplied by another config value that depends on which heating element is fitted to the specific heater to get kWh.

Maybe your heater is similar, in which case this should be changed accordingly.